charts/opentelemetry-collector/templates/_helpers.tpl

```diff
@@ -153,13 +153,13 @@ Return if ingress is stable.
 {{- end }}
 
 {{- define "opentelemetry-collector.annotations" -}}
-{{- if eq .Values.mode "deployment" }}
-{{- if .Values.annotations }}
+{{- if or .Values.annotations .Values.standaloneCollector.annotations -}}
 annotations:
+{{- if .Values.annotations }}
 {{- .Values.annotations | toYaml | nindent 2  }}
 {{- end }}
-{{- else if .Values.standaloneCollector.annotations }}
-annotations:
+{{- if .Values.standaloneCollector.annotations }}
 {{- .Values.standaloneCollector.annotations | toYaml | nindent 2  }}
```

Comment:
regarding `podLabels` it could be consolidated into one function making it easier to maintain
```
{{- define "opentelemetry-collector.daemonset-podLabels" -}}
{{- if eq .Values.mode "daemonset" }}
  {{- if .Values.podLabels }}
  {{- .Values.podLabels | toYaml }}
  {{- end }}
{{- else if .Values.agentCollector.podLabels }}
{{- .Values.agentCollector.podLabels | toYaml }}
{{- end }}
{{- end }}

{{- define "opentelemetry-collector.deployment-podLabels" -}}
{{- if eq .Values.mode "deployment" }}
  {{- if .Values.podLabels }}
  {{- .Values.podLabels | toYaml }}
  {{- end }}
{{- else if .Values.standaloneCollector.podLabels }}
{{- .Values.standaloneCollector.podLabels | toYaml }}
{{- end }}
{{- end }}
```
==>
```
{{- define "opentelemetry-collector.podLabels" -}}
{{- if or .Values.podLabels .Values.agentCollector.podLabels -}}
{{- .Values.podLabels | toYaml }}
{{- else }}
{{- .Values.agentCollector.podLabels | toYaml }}
{{- end }}
{{- end }}
```

Reply:
helm doesn't know how to short circuit right?  In the consolidated template, if `.Values.agentCollector.podLabels` is truthy but `.Values.podLabels` is falsy won't it pass the if condition and then use `.Values.podLabels` empty value?  Also it looks like this breaks backwards compatibility for `.Values.standaloneCollector.podLabels`

Reply:
Sorry, I missed one line to check which was set... but after rereading all the comments I came up with the [change](https://github.com/open-telemetry/opentelemetry-helm-charts/pull/212/commits/97fd6f0ea62dca35daf037276df46952a5b91fd8)

Which consolidates labels and annotations following:
- only use `podLabels`, `podAnnotaions` and `annotations` if `mode` is set with any value (not an empty string)
- only use `standaloneCollector.podLabels`, `standaloneCollector.podAnnotations` and `standaloneCollector.annotations` if `standaloneCollector.enabled = true`